src/core/ScreenLockListenerDBus.cpp

```diff
@@ -0,0 +1,70 @@
+#include "ScreenLockListenerDBus.h"
+
+#include <QDBusConnection>
+#include <QDBusInterface>
+#include <QDBusReply>
+
+ScreenLockListenerDBus::ScreenLockListenerDBus(QWidget *parent):
+    ScreenLockListenerPrivate(parent)
+{
+    QDBusConnection sessionBus = QDBusConnection::sessionBus();
+    QDBusConnection systemBus = QDBusConnection::systemBus();
+
+    sessionBus.connect(
+                "org.freedesktop.ScreenSaver", // service
+                "/org/freedesktop/ScreenSaver", // path
+                "org.freedesktop.ScreenSaver", // interface
+                "ActiveChanged", // signal name
+                this, //receiver
+                SLOT(freedesktopScreenSaver(bool)));
+    
+    sessionBus.connect(
+                "org.gnome.SessionManager", // service
+                "/org/gnome/SessionManager/Presence", // path
+                "org.gnome.SessionManager.Presence", // interface
+                "StatusChanged", // signal name
+                this, //receiver
+                SLOT(gnomeSessionStatusChanged(uint)));
+
+    systemBus.connect(
+                "org.freedesktop.login1", // service
+                "/org/freedesktop/login1", // path
+                "org.freedesktop.login1.Manager", // interface
+                "PrepareForSleep", // signal name
+                this, //receiver
+                SLOT(logindPrepareForSleep(bool)));
+
+    sessionBus.connect(
+                "com.canonical.Unity", // service
+                "/com/canonical/Unity/Session", // path
+                "com.canonical.Unity.Session", // interface
+                "Locked", // signal name
+                this, //receiver
+                SLOT(unityLocked()));
+}
+
+void ScreenLockListenerDBus::gnomeSessionStatusChanged(uint status)
+{
+    if (status != 0) {
```

Comment:
@TheZ3ro ok so from what I can see from gnome's doc :
```
The status parameter must be one of the following:

0: Available
1: Invisible
2: Busy
3: Idle
```

Should we lock for all of the above statuses except for Available (Busy, for instance).

Also, We could extract 0 as a constant to make it more clear why we are ignoring this status : `if (status != gnomeStatusAvailable)`